### PR TITLE
Tensorboard

### PR DESCRIPTION
**Pull request by GucciGui**
Tracking reward and loss as metrics to compare the training progress using TensorBoardX

### CODE REVIEW

File: agent_code/auto_bomber/model.py [hunk 1]
```diff
@@ -32,19 +34,27 @@ def select_best_action(self, game_state: dict, agent_self):
...
                 q_estimations = np.sum(x_all_t * self.weights[action_id], axis=0)
                 residuals = (y_all_t - q_estimations[:, np.newaxis])
+                mean_residuals = np.mean(residuals)
+                loss.append(mean_residuals)
                 q_grad = np.sum(x_all_t.transpose() * residuals, axis=1)
 
                 weight_updates = config.LEARNING_RATE / y_all_t.shape[0] * q_grad
                 self.weights[action_id] += weight_updates
 
+        loss = np.mean(loss)
```
Comment (tkrieger) [inline agent_code/auto_bomber/model.py]: This redefines/reassigns the variable loss with a other shape/type or? (Before it was a list now a scalar or?) Thus I would recommend to use another var name

File: agent_code/auto_bomber/model.py [hunk 2]
```diff
@@ -32,19 +34,27 @@ def select_best_action(self, game_state: dict, agent_self):
         choice = np.random.choice(top_3_actions, p=[0.9, 0.05, 0.05])
         return config.ACTIONS[choice]
 
-    def fit_model_with_transition_batch(self, transitions: Transitions):
+    def fit_model_with_transition_batch(self, transitions: Transitions, round: int):
+        loss = []
```
Comment (tkrieger) [inline agent_code/auto_bomber/model.py:38]: [Optional] The size is known here, thus we can create a fixed size numpy array or?

**Review by tkrieger — COMMENTED** [review]
Only some minor stuff

File: agent_code/auto_bomber/model.py (hunk 2 above)
Comment (GucciGui) [reply agent_code/auto_bomber/model.py:38]: I don't think the size is known previously, as it depends on whether x_all_t.size is zero, right?
Because in case it is zero, it does not enter the if-statement

**Review by PrimeF — APPROVED** [review]
LGTM!

File: agent_code/auto_bomber/model.py (hunk 2 above)
Comment (tkrieger) [reply agent_code/auto_bomber/model.py:38]: yep you are correct, oversaw the if statement
